addons/escoria-core/game/core-scripts/esc_controller.gd

```diff
@@ -0,0 +1,244 @@
+# This script contains functions called by gamedev's game.gd.
+# These functions convert input actions into game actions. For example:
+# "click on background" -> player walks to position
+# "click on item" -> player walks to item then performs the action defined
+# by current verb
+
+class_name ESCController
+
+# Makes an object walk to a destination. This can be either a 2D position or
+# another object.
+#
+# #### Parameters
+#
+# - moving_obj_id: global id of the object that needs to move
+# - destination: Position2D or String of the object the moving object has head
+# to
+# - is_fast: if true, the walk is performed at fast speed (defined in the moving
+# object.
+func perform_walk(moving_obj: ESCObject, destination, is_fast: bool = false):
```

Comment:
You missed changing this to use Position2D or ESCObjects for the destination. Maybe that wasn't clear enough in my last review.

Reply:
I think I missed it, sorry about this. Not sure how to achieve though: I can't use an ESCObject to hold either an ESCItem or a Vector2 because the node data has to be a Node, so I'd need to create a Position2D node just to hold the Vector2 data. I don't think it is worth...

Reply:
No, I meant support an Position2D or an ESCObject, but no string. This way you can just test for one of the two classes (either using `is` or `has_method` or something) and don't need to check the object in the object manager etc.

Reply:
Done